Doc: create style-guide front for internationalization (#21)
GeekHunter-Brasil/hello-world: frontend-styleguide/README.md

```

[Back to top ⬆️](#pushpin-summary)

## Translate

### 👉 Use I18n to translate

> Our plataform has support for internationalized (i18n). First step is configurated [nextJs](https://nextjs.org/docs/advanced-features/i18n-routing) to support locales on the routes.

Reviewer: ```suggestion
Our plataform has support for internationalization (i18n). First step is configurate [nextJs](https://nextjs.org/docs/advanced-features/i18n-routing) to support locales on the routes.
```

Reviewer 2: Should be `configure`, not `configurate`

Author: Fix into `678a506`

Second step, is configure lib [next-translate](https://github.com/vinissimus/next-translate), the plugin is used to render keys into the pages on application.

file `next.config.js`
```js
const nextTranslate = require('next-translate');

module.exports = nextTranslate({
  i18n: {
    locales: ['pt-BR', 'en'],
    defaultLocale: 'pt-BR',
  },
});

> ```

Reviewer 2: I wouldn't break this into first & second steps, since they area already configured into the applications and we don't need to do this regularly or ever again (unless we create another repo).

Maybe it's better to just say that we use nextjs built-in i18n and next-translate to support rendering tokens on pages.

### 👉 Directory Locales structure

> When organizing locale files you must insert each file into the translation folder, thus not creating too big files, with this structure we obtain a better organization.

Author: Fix into `678a506`

```
📁 locales/
└──📁 pt-BR/
   └──📝 page_1.yml
   └──📝 page_2.yml
└──📁 en/
   └──📝 page_1.yml
   └──📝 page_2.yml
```

> Each file inside contains its keys at the same level, there is no need to chain keys .

Reviewer 2: Useless ` `

```suggestion
Each file inside contains its keys at the same level, there is no need to chain keys.
```

Author: Fix into `678a506`

ex: `locales/pt-BR/page_1.yml`.
```yml
example: Olá mundo
```

ex: `locales/en/page_1.yml`.
```yml
example: Hello world

> ```
> ### 👉 Translate page

Reviewer 2: Missing line break

```suggestion
```

### 👉 Translate page
```

> Use the translations in the page:

Reviewer 2: ```suggestion
Use the translations on the page:
```

Author: Fix into `678a506`

```tsx

> import useTranslation from 'next-translate/useTranslation';

Author: Fix into `678a506`

> const Page = (): React.ReactElement => {
>   const { t } = useTranslation('page_1');
> 
>   return (
>     <div>{t('example')}</div>
>   )
> }
> 
> export default Page;

Reviewer 2: Should follow the styleguide on exporting components

```suggestion
export const Page = (): React.ReactElement => {
  const { t } = useTranslation('page_1');

  return (
    <div>{t('example')}</div>
  )
}
```

Author: Fix into `678a506`
